lib/Wasmtime.hs

```diff
@@ -1202,6 +1250,142 @@ data MemoryAccessError = MemoryAccessError deriving (Show)
 
 instance Exception MemoryAccessError
 
+--------------------------------------------------------------------------------
+-- Globals
+--------------------------------------------------------------------------------
+
+-- | The type of a global.
+newtype GlobalType = GlobalType {unGlobalType :: ForeignPtr C'wasm_globaltype_t}
+
+withGlobalType :: GlobalType -> (Ptr C'wasm_globaltype_t -> IO a) -> IO a
+withGlobalType = withForeignPtr . unGlobalType
+
+-- | Returns a new 'GlobalType' with the kind of the given Haskell type and the
+-- specified 'Mutability'.
+newGlobalType :: HasKind a => Proxy a -> Mutability -> IO GlobalType
+newGlobalType proxy mutability = do
+  globaltype_ptr <- newGlobalTypePtr proxy mutability
+  GlobalType <$> newForeignPtr p'wasm_globaltype_delete globaltype_ptr
+
+newGlobalTypePtr :: HasKind a => Proxy a -> Mutability -> IO (Ptr C'wasm_globaltype_t)
+newGlobalTypePtr proxy mutability = do
+  valtype_ptr <- c'wasm_valtype_new $ kind proxy
+  c'wasm_globaltype_new valtype_ptr $ toWasmMutability mutability
+
+-- TODO: think about whether we should reflect Mutability on the type-level
+-- such that we can only use `globalSet` on mutable globals.
+
+-- | Specifies wether a global can be mutated or not.
+data Mutability
+  = -- | The global can not be mutated.
+    Immutable
+  | -- | The global can be mutated via 'globalSet'.
+    Mutable
+  deriving (Show, Eq)
+
+toWasmMutability :: Mutability -> C'wasm_mutability_t
```

Comment:
Yes there is a way and it's called [unsafeCoerce](https://hackage.haskell.org/package/base-4.18.0.0/docs/Unsafe-Coerce.html#v:unsafeCoerce).

Although that wouldn't work in this case since the runtime representation of these types are very likely different.